vcd/datasource_vcd_rde_behavior_invocation_test.go

```diff
@@ -0,0 +1,121 @@
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", configText)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testAccProviders,
+		CheckDestroy:      testAccCheckRdeInterfaceDestroy("vcd_rde_interface.interface"), // If the interface is destroyed, everything is
+		Steps: []resource.TestStep{
+			{
+				Config: configText,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("vcd_rde.rde", "name", t.Name()),
+					// No-op operations return the original entity data, hence the RDE Type should appear:
+					resource.TestMatchResourceAttr("data.vcd_rde_behavior_invocation.invoke", "result",
+						regexp.MustCompile(fmt.Sprintf("\"urn:vcloud:type:%s:%s:%s\"", params["Vendor"], params["Nss"], params["Version"]))),
+					resource.TestMatchResourceAttr("data.vcd_rde_behavior_invocation.invoke2", "result",
+						regexp.MustCompile(fmt.Sprintf("\"urn:vcloud:type:%s:%s:%s\"", params["Vendor"], params["Nss"], params["Version"]))),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccVcdRdeBehaviorInvocation = `
+resource "vcd_rde_interface" "interface" {
+  nss     = "{{.Nss}}"
+  version = "{{.Version}}"
+  vendor  = "{{.Vendor}}"
+  name    = "{{.Name}}"
+}
+
+resource "vcd_rde_interface_behavior" "behavior" {
+  rde_interface_id = vcd_rde_interface.interface.id
+  name             = "{{.Name}}"
+  description      = "{{.Description}}"
+  execution = {
+    "id" : "{{.ExecutionId}}1"
+    "type" : "{{.ExecutionType}}"
+  }
+}
+
+resource "vcd_rde_type" "type" {
+  nss           = "{{.Nss}}"
+  version       = "{{.Version}}"
+  vendor        = "{{.Vendor}}"
+  name          = "{{.Name}}"
+  description   = "{{.Description}}"
+  interface_ids = [vcd_rde_interface.interface.id]
+  schema        = file("{{.SchemaPath}}")
+
+  # Behaviors can't be created after the RDE Interface is used by a RDE Type
+  # so we need to depend on the Behaviors to wait for them to be created first.
+  depends_on = [vcd_rde_interface_behavior.behavior]
+}
+
+resource "vcd_rde" "rde" {
+  org          = "System" # We use System org to avoid using right bundles
+  rde_type_id  = vcd_rde_type.type.id
+  name         = "{{.Name}}"
+  resolve      = true
+  input_entity = file("{{.EntityPath}}")
+}
+
+# Required Access Levels to invoke Behaviors
+resource "vcd_rde_type_behavior_acl" "interface_acl" {
+  rde_type_id = vcd_rde_type.type.id
+  behavior_id = vcd_rde_interface_behavior.behavior.id
+  access_level_ids = [{{.AccessLevels}}]
+}
+
+data "vcd_rde_behavior_invocation" "invoke" {
+  rde_id                  = vcd_rde.rde.id
+  behavior_id             = vcd_rde_interface_behavior.behavior.id
+}
+
+data "vcd_rde_behavior_invocation" "invoke2" {
+  rde_id                  = vcd_rde.rde.id
+  behavior_id             = vcd_rde_interface_behavior.behavior.id
+  arguments = {
+    "arg1" : "not_used"
+  }
+  metadata = {
+    "meta1" : "not_used"
+  }
+}
+`
```

Comment:
Maybe it would be useful to have in the test output, which would check that the result is there too?

Reply:
Actually the output (the `result` attribute) is checked on lines 48-51, do you mean a different thing?

Reply:
I have missed, it's all good then, thanks!